merkle/tests/tree_dup_odd_leaf_merger.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+// Test cases are generated from ocaml code
+//
+// (* utop -require core_kernel *)
+// #use "https://github.com/o1-labs/snarky/blob/master/src/base/merkle_tree.ml"
+//
+// let my_hash x = x
+
+// let my_merge a b =
+//   match a with
+//   | None -> None
+//   | Some a -> ( match b with None -> Some (a * 2) | Some b -> Some (a + b))
+
+// let create_tree n =
+//   let tree = create ~hash:my_hash ~merge:my_merge 0 in
+//   add_many tree (List.init (n - 1) ~f:(fun i -> i + 1))
+//
+// create_tree 127 |> root
+// create_tree 127 |> depth
+
+#[cfg(test)]
+mod tests {
+    use mina_merkle::*;
+
+    struct TestHasher {}
+
+    impl MerkleHasher<2> for TestHasher {
+        type Item = i64;
+        type Hash = i64;
+        fn hash(item: &Self::Item, _: MerkleTreeNodeMetadata<2>) -> Self::Hash {
+            *item
+        }
+    }
+
+    struct TestMerger {}
+
+    impl MerkleMerger<2> for TestMerger {
+        type Hash = i64;
+        fn merge(
+            items: [&Option<(Self::Hash, MerkleTreeNodeMetadata<2>)>; 2],
+        ) -> Option<Self::Hash> {
+            if let Some((left, _)) = items[0] {
+                if let Some((right, _)) = items[1] {
+                    Some(left + right)
+                } else {
+                    Some(*left * 2)
+                }
+            } else {
+                None
+            }
```

Comment:
A bit more compact and readable if you like

```suggestion
            match (items[0], items[1]) {
                (Some((left, _)), Some((right, _))) => Some(left+right),
                (Some((left, _)), None)) => Some(*left * 2),
                _ => None,
            }
```

Reply:
Fixed, thx!